src/main/java/tc/oc/pgm/rotation/FixedPGMMapOrderManager.java

```diff
@@ -0,0 +1,225 @@
+  }
+
+  public FixedPGMMapOrder getActiveRotation() {
+    return activeRotation;
+  }
+
+  void setEvaluatingPlayerCount(boolean evaluatingPlayerCount) {
+    this.isEvaluatingPlayerCount = evaluatingPlayerCount;
+  }
+
+  public boolean isEvaluatingPlayerCount() {
+    return isEvaluatingPlayerCount;
+  }
+
+  public List<FixedPGMMapOrder> getRotations() {
+    return rotations;
+  }
+
+  public void saveCurrentPosition() {
+    FileConfiguration rotationFileConfiguration =
+        YamlConfiguration.loadConfiguration(rotationsFile);
+
+    rotationFileConfiguration.set(
+        activeRotation.getName() + ".next_map",
+        activeRotation.getMapInPosition(activeRotation.getPosition() + 1).getName());
+
+    try {
+      rotationFileConfiguration.save(rotationsFile);
+    } catch (IOException e) {
+      logger.log(
+          Level.SEVERE, "Could not save current rotation's position for future reference", e);
+    }
+  }
+
+  public void recalculateActiveRotation() {
+    int activePlayers = getActivePlayers();
+    List<Integer> playerCounts = new ArrayList<>(rotations.size());
+
+    rotations.forEach(
+        rotation -> {
+          if (rotation.isEnabled()) {
+            playerCounts.add(rotation.getPlayers());
+          }
+        });
+
+    Collections.sort(playerCounts);
+    FixedPGMMapOrder newRotation = null;
+
+    int count = 0;
+    while (count < playerCounts.size()) {
+      newRotation = getRotationByPlayerCount(playerCounts.get(count));
+
+      if (playerCounts.get(count) >= activePlayers) {
+        break;
+      }
+
+      count++;
+    }
+
+    updateActiveRotation(newRotation);
+  }
+
+  private int getActivePlayers() {
+    Collection<? extends Player> onlinePlayers = Bukkit.getOnlinePlayers();
+    if (onlinePlayers.isEmpty()) return 0;
+
+    Match match = matchManager.getMatch((Player) onlinePlayers.toArray()[0]);
+
+    assert match != null;
+    return onlinePlayers.size() - match.getObservers().size() / 2;
+  }
+
+  private void updateActiveRotation(FixedPGMMapOrder rotation) {
+    if (rotation == activeRotation) return;
+
+    setActiveRotation(rotation);
+    Bukkit.broadcastMessage(
+        ChatColor.WHITE
+            + "["
+            + ChatColor.GOLD
+            + "Rotations"
+            + ChatColor.WHITE
+            + "] "
+            + ChatColor.GREEN
+            + AllTranslations.get()
+                .translate(
+                    "rotations.rotationChange",
+                    Bukkit.getConsoleSender(),
+                    (ChatColor.AQUA + rotation.getName() + ChatColor.GREEN)));
+  }
+
+  private FixedPGMMapOrder getRotationByPlayerCount(int playerCount) {
+    AtomicReference<FixedPGMMapOrder> matchByPlayerCount = new AtomicReference<>();
+
+    rotations.forEach(
+        rotation -> {
+          if (rotation.getPlayers() == playerCount) {
+            matchByPlayerCount.set(rotation);
+          }
+        });
+
+    return matchByPlayerCount.get();
```

Comment:
Using an atomic reference to edit a value from inside a stream, is a bad idea. 
If you want to do the edit-variable loop, do a normal `for (FixedPgmMapOrder rot : rotations) {`
If you wanna do lambda-style, do make a proper stream:
```java
return rotations.stream()
    .filter(rotation -> rotation.getPlayers() == playerCount)
    .findFirst().orElse(null);
```